pytest_inmanta/plugin.py

```diff
@@ -1320,9 +1321,18 @@ def enumerate_model(model: str):
                     DeprecationWarning,
                 )
 
+        # Fix the resource serialization (fully serialize them)
+        new_resources = {}
+        for resource_id, resource in self.resources.items():
```

Comment:
That will not work, you are accessing the resources before setting them (you are actually reading the resources from the previous compile)